### PR TITLE
Fix for http protocol check for object storage endpoint

### PR DESCRIPTION
**Pull request by bvtejaswi**
### :nut_and_bolt: Description: What code changed, and why?
As a user I need to provide object storage endpoint with http/https in config.toml but this check will give error which is not valid

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-3735

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Before this change
Screenshot 2023-06-27 at 7 34 09 PM
After this change it moved forward
Screenshot 2023-06-27 at 7 35 20 PM

### CODE REVIEW

File: lib/config/helper_test.go [hunk 1]
```diff
@@ -364,29 +364,19 @@ func TestValidateS3Endpoint(t *testing.T) {
 	}{
 		{
 			name:          "Valid Endpoint",
-			endpoint:      "s3-external-1.amazonaws.com",
-			expectedError: false,
+			endpoint:      "",
```
Comment (vivek-yadav) [inline lib/config/helper_test.go:367]: why did you remove the value ?
Just expect error should be changed to false ?

Comment (bvtejaswi) [reply lib/config/helper_test.go:367]: its the unit test check that if endpoint is empty then it will throw an error
